writer.go

```diff
@@ -19,7 +19,25 @@ type Writer struct {
 	framer    Framer
 	formatter Formatter
 
-	conn serverConn
+	conn          serverConn
+	needReconnect bool
+}
+
+// Poll the socket to find out if/when it closes
+func (w *Writer) poll() {
+	for {
+		if w.conn == nil {
+			w.needReconnect = true
+			return
+		}
+
+		b := make([]byte, 1)
+		_, err := w.conn.read(b)
```

Comment:
There's no delay within this `for` loop, meaning that we're going to slam the network with reads as fast as we possibly can and fill up a CPU core. We don't want to do that.